src/components/BankTransactionsList/BankTransactionsListItem.tsx

```diff
@@ -219,49 +201,48 @@ export const BankTransactionsListItem = ({
       <span className='Layer__bank-transaction-list-item__expanded-row' onClick={preventRowExpansion}>
         <AnimatedPresenceDiv variant='expand' isOpen={openExpandedRow} key={`expanded-${bankTransaction.id}`}>
           <ExpandedBankTransactionRow
-            ref={expandedRowRef}
             bankTransaction={bankTransaction}
             isOpen={openExpandedRow}
-            close={() => setOpenExpandedRow(false)}
             categorized={displayAsCategorized}
             asListItem={true}
             submitBtnText={
               displayAsCategorized
                 ? stringOverrides?.updateButtonText || 'Update'
                 : stringOverrides?.approveButtonText || 'Approve'
             }
-            containerWidth={containerWidth}
 
             showDescriptions={showDescriptions}
             showReceiptUploads={showReceiptUploads}
             showTooltips={showTooltips}
 
             variant='list'
+            onValidityChange={setIsExpandedRowValid}
           />
         </AnimatedPresenceDiv>
       </span>
-      {!openExpandedRow && categorizationEnabled && !displayAsCategorized && (
+      {categorizationEnabled && !displayAsCategorized && (
         <div onClick={preventRowExpansion}>
-          <HStack pi='md' gap='md' pb='md'>
-            <BankTransactionCategoryComboBox
-              bankTransaction={bankTransaction}
-              selectedValue={selectedCategory ?? null}
-              onSelectedValueChange={(selectedCategory: BankTransactionCategoryComboBoxOption | null) => {
-                setTransactionCategory(bankTransaction.id, selectedCategory)
-                setShowRetry(false)
-              }}
-              isDisabled={bankTransaction.processing}
-            />
+          <HStack pi='md' gap='md' pb='md' justify='end'>
+            {!openExpandedRow && (
+              <BankTransactionCategoryComboBox
+                bankTransaction={bankTransaction}
+                selectedValue={selectedCategory ?? null}
+                onSelectedValueChange={(selectedCategory: BankTransactionCategoryComboBoxOption | null) => {
+                  setTransactionCategory(bankTransaction.id, selectedCategory)
+                }}
+                isDisabled={isProcessing}
+              />
+            )}
             <SubmitButton
-              disabled={bankTransaction.processing}
-              onClick={handleSave}
-              className={showRetry ? 'Layer__bank-transaction__retry-btn' : 'Layer__bank-transaction__submit-btn'}
-              processing={bankTransaction.processing}
+              disabled={isProcessing}
+              onClick={() => { void save() }}
```

Comment:
As compared to the `BankTransactionTable` or a row, I think this `<SubmitButton>` is fine because a submit button is so small that re-rendering it (because the callback params change) is negligible, trivial, inexpensive.